src/main/java/hudson/tasks/junit/SuiteResult.java

```diff
@@ -348,4 +349,19 @@ void setParent(hudson.tasks.junit.TestResult parent) {
     private static final long serialVersionUID = 1L;
 
     private static final Pattern SUREFIRE_FILENAME = Pattern.compile("TEST-(.+)\\.xml");
+
+    /**
+     * Merges another SuiteResult into this one.
+     * 
+     * @param sr
+     */
+    public void merge(SuiteResult sr) {
+        if (hasTimeAttr) {
```

Comment:
Would be resolved by test cases checking

```xml
<testsuites name="Automation Tests" tests="2" errors="0" failures="0" ignored="0">
    <testsuite name="test.fs.FileSystemTests">
     <testcase name="testPrefix1" classname="test.fs.FileSystemTest1" time="10"/>
    </testsuite>
    <testsuite name="test.fs.FileSystemTests" time="50">
      <testcase name="testPrefix2" classname="test.fs.FileSystemTest1" time="10"/>
    </testsuite>
</testsuites>
```

and

```xml
<testsuites name="Automation Tests" tests="2" errors="0" failures="0" ignored="0">
    <testsuite name="test.fs.FileSystemTests" time="50">
     <testcase name="testPrefix1" classname="test.fs.FileSystemTest1" time="10"/>
    </testsuite>
    <testsuite name="test.fs.FileSystemTests">
      <testcase name="testPrefix2" classname="test.fs.FileSystemTest1" time="10"/>
    </testsuite>
</testsuites>
```

both of which I would expect to report a total time of 60. (Right?)